flink-core/src/main/java/org/apache/flink/core/fs/FileSystem.java

```diff
@@ -527,13 +527,30 @@ public static FileSystem getUnguardedFileSystem(final URI fsUri) throws IOExcept
                 try {
                     fs = FALLBACK_FACTORY.create(uri);
                 } catch (UnsupportedFileSystemSchemeException e) {
-                    throw new UnsupportedFileSystemSchemeException(
-                            "Could not find a file system implementation for scheme '"
-                                    + uri.getScheme()
-                                    + "'. The scheme is not directly supported by Flink and no Hadoop file system to "
-                                    + "support this scheme could be loaded. For a full list of supported file systems, "
-                                    + "please see https://nightlies.apache.org/flink/flink-docs-stable/ops/filesystems/.",
-                            e);
+                    if (DIRECTLY_SUPPORTED_FILESYSTEM.containsKey(uri.getScheme())) {
+                        final Collection<String> plugins =
+                                DIRECTLY_SUPPORTED_FILESYSTEM.get(uri.getScheme());
+                        throw new UnsupportedFileSystemSchemeException(
+                                String.format(
+                                        "Could not find a file system implementation for scheme '%s'. The scheme is "
+                                                + "directly supported by Flink through the following plugin%s: %s "
+                                                + "and no Hadoop file system to support this scheme could be loaded. Please ensure that each "
+                                                + "plugin resides within its own subfolder within the plugins directory. See https://ci.apache"
+                                                + ".org/projects/flink/flink-docs-stable/ops/plugins.html for more information."
```

Comment:
The documentation is no longer hosted at ci.apache.org, it would be good to directly point to the correct URL.